sansio_lsp_client/events.py

```diff
@@ -1,90 +1,80 @@
 import typing as t
 
-import attr
-from attr import attrs, attrib
+from pydantic import BaseModel, PrivateAttr
 
 if t.TYPE_CHECKING:  # avoid import cycle at runtime
     from .client import Client
 from .structs import (
     JSONDict,
     Diagnostic,
-    Range,
     MessageType,
     MessageActionItem,
     CompletionItem,
     CompletionList,
     TextEdit,
 )
 
+Id = t.Union[int, str]
```

Comment:
When is this a string?

Reply:
I'm not sure if it ever *actually* is, but the spec defines it as `id: integer | string;`